go/kbfs/libfs/profilelist.go

```diff
@@ -56,3 +94,165 @@ func IsSupportedProfileName(name string) bool {
 	// just support alphanumeric filenames.
 	return profileNameRE.MatchString(name)
 }
+
+// ProfileFS provides an easy way to browse the go profiles.
+type ProfileFS struct {
+	config libkbfs.Config
+}
+
+// NewProfileFS returns a read-only filesystem for browsing profiles.
+func NewProfileFS(config libkbfs.Config) ProfileFS {
+	return ProfileFS{config}
+}
+
+var _ libkbfs.NodeFSReadOnly = ProfileFS{}
+
+// Lstat implements the libkbfs.NodeFSReadOnly interface.
+func (pfs ProfileFS) Lstat(filename string) (os.FileInfo, error) {
+	if strings.HasPrefix(filename, CPUProfilePrefix) ||
+		strings.HasPrefix(filename, TraceProfilePrefix) {
+		// Set profile sizes to 0 because it's too expensive to read them
+		// ahead of time.
+		return &wrappedReadFileInfo{filename, 0, time.Now(), false}, nil
+	}
+
+	if !IsSupportedProfileName(filename) {
+		return nil, errors.Errorf("Unsupported profile %s", filename)
+	}
+
+	f := ProfileGet(filename)
+	// Get the data, just for the size.
```

Comment:
Have you thought about just returning 0 for all other profiles as well (not just cpu/trace). The size won't be accurate anyway since by the time read happens they must have changed.

Reply:
Well, that's the same as any other file on the file system.  The size can always change after you look it up, but before you read the file. I think it's better to give a decent-enough number when possible, so the caller can allocate reasonable buffers ahead of time if possible.  But if there are problems with that approach, let me know.

Reply:
Got it; sounds good!